docs/core/event_handler/bedrock_agents.md

```diff
@@ -0,0 +1,291 @@
+
+	```python
+    --8<-- "examples/event_handler_bedrock_agents/cdk/bedrock_agent_stack.py"
+	```
+
+	1. The path to your Lambda function handler
+	2. The path to the OpenAPI schema describing your API
+
+### Your first Agent
+
+To create an Agent for Amazon Bedrock, use the `BedrockAgentResolver` to annotate your actions.
+This is similar to the way [all the other Event Handler](api_gateway.md) resolvers work.
+
+=== "Lambda handler"
+
+	The resolvers used by Agents for Amazon Bedrock are compatible with the full suite of Powertools for AWS Lambda utilities.
+	This includes [Logger](../logger.md), [Metrics](../metrics.md) and [Tracer](../tracer.md).
+
+    ```python hl_lines="4 9 12 21"
+    --8<-- "examples/event_handler_bedrock_agents/src/getting_started.py"
+    ```
+
+	1. `description` is a **required** field that should contain a human readable description of your action
+	2. We take care of **parsing**, **validating**, **routing** and **responding** to the request.
+
+=== "OpenAPI schema"
+
+	Powertools for AWS Lambda [generates this automatically](#generating-openapi-schemas) from the Lambda handler.
+
+	```json
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started_schema.json"
+	```
+
+=== "Input payload"
+
+	```json hl_lines="4 6 13"
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started.json"
+	```
+
+=== "Output payload"
+
+	```json hl_lines="10"
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started_output.json"
+	```
+
+!!! note "It's required to include a `description` for each API endpoint and input parameter. This will improve the understanding Amazon Bedrock has of your actions."
+
+### Validating input and output
+
+You can define the expected format for incoming data and responses by using type annotations.
+
+=== "Lambda handler"
+
+	```python hl_lines="5 23-26 32-33"
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started_with_validation.py"
+	```
+
+	1. No need to add the `enable_validation` parameter, as it's enabled by default.
+	2. You can define constraints using standard Python types, [dataclasses](https://docs.python.org/3/library/dataclasses.html) or [Pydantic models](https://docs.pydantic.dev/latest/concepts/models/)
+	3. Describe each input and output using human-readable descriptions
+	4. Add the typing annotations to your parameters and return types, and let the event handler take care of the rest
+
+=== "OpenAPI schema"
+
+	```json
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started_with_validation_schema.json"
+	```
+
+=== "Input payload"
+
+	```json hl_lines="6-13 20"
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started_with_validation.json"
+	```
+
+=== "Output payload"
+
+	```json hl_lines="10"
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started_with_validation_output.json"
+	```
+
+If the request validation, your event handler will not be called, and an error message is returned to Bedrock.
```

Comment:
As suggested earlier, please create an exception handling section with an example output etc. We also discussed having sequence diagrams to demystify this area.

Reply:
I remember there was no concensus on that decision since we don't control what Bedrock does with the output of our APIs. I can add an example payload when the validation fails, but unsure how that helps.